### PR TITLE
fix a IndexOutOfBoundsException from WidgetIndentsHighlightingPass

### PR DESCRIPTION
**Pull request by devoncarew**
- fix a IndexOutOfBoundsException from WidgetIndentsHighlightingPass
- fix https://github.com/flutter/flutter-intellij/issues/4047 (`IndexOutOfBoundsException: Wrong offset: -1. Should be in range: [0, 283]`)

### CODE REVIEW

File: src/io/flutter/editor/WidgetIndentsHighlightingPass.java [hunk 1]
```diff
@@ -135,13 +135,13 @@ boolean updateSelected(@NotNull Editor editor, @NotNull RangeHighlighter highlig
...
 
-      if (startOffset >= doc.getTextLength()) {
+      if (startOffset < 0 || startOffset >= doc.getTextLength()) {
         return setSelection(false);
       }
 
       final int endOffset = highlighter.getEndOffset();
 
-      int off = startOffset;
+      int off;
```
Comment (jacob314) [inline src/io/flutter/editor/WidgetIndentsHighlightingPass.java]: was this change intentional? Seems accidental.

Comment (devoncarew) [reply src/io/flutter/editor/WidgetIndentsHighlightingPass.java]: It was - IntelliJ helpfully told me that `off` was assigned to again before it was first read, so this initial value was never used.

**Comment by jacob314** [comment]
lgtm
